qiskit/quantum_info/states/utils.py

```diff
@@ -60,6 +60,9 @@ def partial_trace(state, qargs):
         return DensityMatrix(rho, dims=traced_shape._dims_l)
 
     # Density matrix case
+    # Empty partial trace case.
+    if traced_shape == state._op_shape:
```

Comment:
Question: Can this condition be replaced by checking an empty list? I think it is straight forward from above your comment. Sorry if there is some deeper reason.

Reply:
Sure, it's marginally more efficient to do it your way as well, especially if the shapes are huge.  Done in 1e9e40b.